### DISCUSSION TITLE
Is it safe to link a static lib MSVC build to a program that dynamically links the C runtime?

### DISCUSSION THREAD

**Discussion by ts826848** [jedisct1/libsodium#1339] [Q&A]
I'm working on integrating libsodium into a Windows program and noticed that the static library build statically links to the C runtime. The program itself dynamically links against the C runtime, so if I use the static library I think this will result in the program running with 2 C runtimes - 1 statically linked into libsodium and 1 for the rest of the program, which dynamically links to the C runtime.

Is this setup generally safe for use with libsodium? I currently use libsodium's signature verification and base64 functions, and I think those read/write into user-provided buffers, so I don't think those would run into the main pitfalls described in [this comment](https://github.com/jedisct1/libsodium/issues/614#issuecomment-337766603). Are there any other potential issues I should be aware of?

**Comment by jedisct1** [comment]
C has no runtime, so it should be fine.

**Reply by ts826848** [reply]
Sorry for the delay - seems I missed the email from GitHub that you replied.

I'm a bit confused by your response - MSVC definitely [links against a C runtime](https://learn.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features?view=msvc-170) and the comment I linked discussed the implications of that for libsodium. I'm at a bit of a loss about how to interpret your response :(
